pipelines/controller/src/main/java/com/google/fhir/analytics/ApiController.java

```diff
@@ -166,4 +178,60 @@ public ResponseEntity<String> testView(
     }
     return new ResponseEntity<>(response, status);
   }
+
+  @GetMapping("/next")
+  public ScheduleDto getNextScheduled() {
+    ScheduleDto schedule = new ScheduleDto();
+    LocalDateTime nextRun = pipelineManager.getNextIncrementalTime();
+    if (nextRun == null) {
+      schedule.setNextRun("NOT SCHEDULED");
+    } else {
+      schedule.setNextRun(nextRun.toString());
+    }
+    return schedule;
+  }
+
+  @GetMapping("/dwh")
+  public DwhDto getDwh() {
+    DwhDto dwhDto = new DwhDto();
+    String dwh = pipelineManager.getCurrentDwhRoot();
+    dwhDto.setDwhPrefix(dataProperties.getDwhRootPrefix());
+    dwhDto.setDwhPath(dwh == null || dwh.isEmpty() ? "" : dwh);
+    dwhDto.setDwhSnapshots(dwhFilesManager.listDwhSnapshots());
+    return dwhDto;
+  }
+
+  @DeleteMapping("/dwh")
+  @ResponseStatus(HttpStatus.NO_CONTENT)
+  public void deleteSnapshot(@RequestParam String snapshotId) {
+    dwhFilesManager.deleteDwhSnapshotFiles(snapshotId);
+  }
+
+  @GetMapping("/config")
+  public Map<String, String> getConfigs() {
+    return getConfigMap(null);
+  }
+
+  @GetMapping("/config/{name}")
+  public Map<String, String> getConfigs(@PathVariable String name) {
+    return getConfigMap(name);
+  }
+
+  private Map<String, String> getConfigMap(String configName) {
+    List<DataProperties.ConfigFields> configParams = dataProperties.getConfigParams();
```

Comment:
Optional: While you are at this, do you mind taking a look at [this TODO](https://github.com/google/fhir-data-pipes/blob/8035ea050ebe7c01d6649c309f45e0f85b54cb5c/pipelines/controller/src/main/java/com/google/fhir/analytics/DataProperties.java#L249) and automate it (e.g., with reflection) if it is not too much extra work?

Reply:
Need to think about this a bit more I think e.g. if I use reflection on the existing properties to generate the list we want to be sure we only expose the private variables we need. Depending on implementation it might take significant refactoring. I'd recommend working making this change as a separate PR.